lib/update/update_donation.rb

```diff
@@ -70,9 +70,9 @@ def self.update_payment(donation_id, data)
       end
     end
 
+    donation = existing_payment.donation
+    ret = donation.attributes
```

Comment:
Maybe move this one inside the transaction block too. I'm not positive it really matters though.